sdk/ai/ai-projects/src/agents/vectorStores.ts

```diff
@@ -58,19 +62,67 @@ export async function listVectorStores(
 export async function createVectorStore(
   context: Client,
   options: CreateVectorStoreOptionalParams = {},
-): Promise<VectorStoreOutput> {
+): Promise<CreateVectorStoreResponse> {
   const createOptions: GeneratedParameters.CreateVectorStoreParameters = {
     ...operationOptionsToRequestParameters(options),
     body: ConvertToWire.convertVectorStoreOptions(options),
   };
-
   validateCreateVectorStoreParameters(createOptions);
-  const result = await context.path("/vector_stores").post(createOptions);
 
-  if (!expectedStatuses.includes(result.status)) {
-    throw createRestError(result);
+  const initialResponse = await context.path("/vector_stores").post(createOptions);
+  if (!expectedStatuses.includes(initialResponse.status)) {
+    throw createRestError(initialResponse);
   }
-  return ConvertFromWire.convertVectorStoreOutput(result.body);
+  
+  const abortController = new AbortController();
+  const poller: RunningOperation<VectorStoreOutput> = {
+    sendInitialRequest: async () => {
+      return {
+        flatResponse: initialResponse,
+        rawResponse: {
+          ...initialResponse,
+          statusCode: Number.parseInt(initialResponse.status),
+          body: ConvertFromWire.convertVectorStoreOutput(initialResponse.body as WireVectorStoreOutput),
+        }
+      }
+    },
+    sendPollRequest: async (_path: string, pollOptions?: { abortSignal?: AbortSignalLike }) => {
+      function abortListener(): void {
+        abortController.abort();
+      }
+      const inputAbortSignal = pollOptions?.abortSignal;
+      const abortSignal = abortController.signal;
+      if (inputAbortSignal?.aborted) {
+        abortController.abort();
+      } else if (!abortSignal.aborted) {
+        inputAbortSignal?.addEventListener("abort", abortListener, {
+          once: true,
+        });
+      }
+      let response;
+      try {
+        const getOptions: GetVectorStoreOptionalParams = {
+          ...operationOptionsToRequestParameters(options),
+        };
+        response = await context
+          .path("/vector_stores/{vectorStoreId}", initialResponse.body.id)
+          .get({ ...getOptions, abortSignal});
+      } finally {
+        inputAbortSignal?.removeEventListener("abort", abortListener);
+      }
+      return {
+        flatResponse: response,
+        rawResponse: {
+          ...response,
+          statusCode: Number.parseInt(response.status),
+          body: ConvertFromWire.convertVectorStoreOutput(response.body as WireVectorStoreOutput),
+        }
+      }
+    },
+  }
+
+  // TODO: Adjust parameters to match new polling options for v3 (intervalInMS and restoreFrom)
+  return createHttpPoller(poller, {intervalInMs: options.pollingOptions?.sleepIntervalInMs, restoreFrom: undefined})
```

Comment:
Is this required ?

Reply:
I show how we can provide optional parameters from user here in most recent commit